### DISCUSSION TITLE
How to mount NFS  from an NFS server container into another container? (Both running rootless Podman)

### DISCUSSION THREAD

**Discussion by eriksjolund** [containers/podman#13982] [Q&A] [answered]
I'm not sure this is even possible, but after reading in the Podman chat channel
https://app.element.io/#/room/!SEjxdEhDPWoPxzjWCX:matrix.org/$TGOvx5IcbWYXmgT4p5NkM8nnETk92-pPrGgeulI51wE

```
podman volume create --opt type=nfs --opt o=async --opt device=mynfsserver:/path/to/share
neat that it works rootless now
```

and also some discussion in the linux-nfs mailing list
https://lore.kernel.org/linux-nfs/3ee50563b755cd5c1ecfa5be0c6127f5a52e9d16.camel@hammerspace.com/

quote: _When using AUTH_GSS, the knfsd server doesn't care about the
originating port, which would allow unprivileged mounts to go ahead
provided that the user specifies the 'noresvport' mount option on
the client. Isn't that working?_

I get some hope that it should be possible.

There is also [NFS-Ganesha](https://github.com/nfs-ganesha/nfs-ganesha), an NFS server that is running in user mode.

This web page looks to be related to Kerberos:
https://github.com/nfs-ganesha/nfs-ganesha/wiki/RPCSEC_GSS

This seems like a fun project to explore: Running rootless Podman and let one container somehow mount NFS from an NFS-Ganesha container.

Please tell me if the endeavour is blocked by some showstoppers (e.g. missing functionality).

**Comment by rhatdan** [comment, marked as answer]
I know of no reason this would be blocked, other then the client container being allowed to mount a file system.

**Comment by concatime** [comment]
@eriksjolund, were you able to do it finally? An update would be welcome!

**Reply by eriksjolund** [reply]
@concatime No, I'm a bit lost here. I'm not sure it's even possible.

**Reply by concatime** [reply]
Is there a way to mount NFS with udiskctl? Ultimately, I would like to have the ability to mount NFS w/o root (changing /etc/fstab is not a viable solution because I don't know the remote disk in advance).
1. Mount on host w/o root (udisksctl?)
2. Create a volume that binds to the mounted path
3. Start a rootless container with the new volume mounted

SSHFS can be used w/o root, SAMBA too I think. But requiring root to mount a NFS is a real PITA.
